File: api/spring-boot-api/src/main/java/com/spicytodo/dto/TodoCreateRequest.java
package com.spicytodo.dto;

import com.spicytodo.model.Priority;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class TodoCreateRequest {
    @NotBlank(message = "Todo text is required")
    @Size(min = 1, max = 500, message = "Todo text must be between 1 and 500 characters")
    private String text;
    
    private Priority priority;
    private Boolean completed;
    private String dueDate;
    private String reminderTime;
}

File: api/dotnet-api/Models/TodoCreateRequest.cs
using System.ComponentModel.DataAnnotations;

namespace DotnetApi.Models
{
    public class TodoCreateRequest
    {
        [Required(ErrorMessage = "Todo text is required")]
        [StringLength(500, MinimumLength = 1, ErrorMessage = "Todo text must be between 1 and 500 characters")]
        public string Text { get; set; } = string.Empty;
        
        public Priority? Priority { get; set; }
        public bool? Completed { get; set; }
        public string? DueDate { get; set; }
        public string? ReminderTime { get; set; }
    }
}

